src/pytorch_lightning/trainer/trainer.py

```diff
@@ -961,7 +961,7 @@ def _restore_modules_and_callbacks(self, checkpoint_path: Optional[_PATH] = None
     def _run(
         self, model: "pl.LightningModule", ckpt_path: Optional[str] = None
     ) -> Optional[Union[_EVALUATE_OUTPUT, _PREDICT_OUTPUT]]:
-        if self.state.fn in (TrainerFn.FITTING, TrainerFn.TUNING):
+        if self.state.fn == TrainerFn.FITTING:
```

Comment:
there's no way to determine tuning now

Reply:
That's problematic, because the logic here is guarding code that should not execute when optimizing parameters.

Reply:
I don't understand how it is possible to suddenly remove this logic just because of a deprecation?

Reply:
with the current logic, on the second call from within the callback, this function won't do anything since all of these values will be set when `.fit` is called.